test/integration/revocation_test.go

```diff
@@ -167,3 +170,53 @@ func TestRevokeWithKeyCompromise(t *testing.T) {
 	test.AssertError(t, err, "NewAccount didn't fail with a blacklisted key")
 	test.AssertEquals(t, err.Error(), `acme: error code 400 "urn:ietf:params:acme:error:badPublicKey": public key is forbidden`)
 }
+
+func TestBadKeyRevoker(t *testing.T) {
```

Comment:
This is a nice integration test. I'd like to make it a bit nicer:

- Issue multiple certificates for both cA and cB to exercise the multiple-certificates code paths.
- Introduce a third and fourth client, one with the same email as cB, and one with a different email, so we can test the code path that chunks up emails by recipient.